src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -226,69 +203,20 @@ export class QuickBar extends LitElement {
           "domain",
           domainToName(this.hass.localize, domain)
         ),
-      domain,
-      service: "reload",
+      icon: domainIcon(domain),
+      action: () => this.hass.callService(domain, "reload"),
     }));
   }
 
-  private _filterCommandItems(
-    items: CommandItem[],
-    filter: string
-  ): CommandItem[] {
-    return items
-      .filter(({ text }) =>
-        fuzzySequentialMatch(filter.toLowerCase(), [text.toLowerCase()])
-      )
-      .sort((itemA, itemB) => compare(itemA.text, itemB.text));
-  }
-
-  private _filterEntityItems(filter: string): HassEntity[] {
-    return this._entities
-      .filter(({ entity_id, attributes: { friendly_name } }) => {
-        const values = [entity_id];
-        if (friendly_name) {
-          values.push(friendly_name);
-        }
-        return fuzzySequentialMatch(filter.toLowerCase(), values);
-      })
-      .sort((entityA, entityB) =>
-        compare(entityA.entity_id, entityB.entity_id)
-      );
-  }
-
-  private async _processCommand(ev: SingleSelectedEvent) {
-    const index = ev.detail.index;
-    const item = (ev.target as any).items[index].item;
-
-    this._commandTriggered = index;
-
-    this._runCommandAndCloseDialog({
-      domain: item.domain,
-      service: item.service,
-      serviceData: item.serviceData,
-    });
-  }
-
-  private async _runCommandAndCloseDialog(request?: ServiceCallRequest) {
-    if (!request) {
-      return;
-    }
-
-    this.hass
-      .callService(request.domain, request.service, request.serviceData)
-      .then(() => this.closeDialog());
-  }
-
-  private _entityMoreInfo(ev: SingleSelectedEvent) {
-    const index = ev.detail.index;
-    const entityId = (ev.target as any).items[index].entityId;
-
-    this._launchMoreInfoDialog(entityId);
-  }
-
-  private _launchMoreInfoDialog(entityId) {
-    fireEvent(this, "hass-more-info", { entityId });
-    this.closeDialog();
+  private _generateEntityItems(): QuickBarItem[] {
+    return Object.keys(this.hass.states).map((entityId) => ({
+      text: computeStateName(this.hass.states[entityId]),
+      altText: entityId,
+      icon: domainIcon(computeDomain(entityId)),
```

Comment:
Let's pass the state to `domainIcon`
```suggestion
      icon: domainIcon(computeDomain(entityId, this.hass.states[entityId])),
```

Reply:
Sure thing. What's that give us?

Looking at the code, it looks like it gives us better specificity on icons based on things like a binary sensor's classification?

Reply:
yep :-)